### ISSUE TITLE
[M95] Limit number of connections per hour and day

### ISSUE THREAD

**Issue by ralcaide** [Sinapse-Energia/AP-Sinapse#184]
In some real installations we are having consumption problems. One of the reasons is that some devices try to connect to the network a lot (maybe) maybe due to a coverage problems or some other problems. As a hot fix we would like to limit the number of connections per hour and per day when GPRS is used. 

The maximum number of connections should be easy to configure in the Definitions.h. Normally we will use something like:

MAX_CONNECTIONS_HOUR = 10
MAX_CONNECTIONS_DAY = 50

Related with #176

**Comment by ralcaide** [comment]
@fpgamcu , the consumption problem could be a huge problem for us. Please, work on this with high priority.

If you have any idea about improvements, please share

**Comment by fpgamcu** [comment]
Estimation is 4 hours. Commit will be ready today.

**Comment by ralcaide** [comment]
great, thanks!

**Comment by ralcaide** [comment]
Thanks, should this behavior be implemented also in Bootloader or there is already limited the number of connections?

**Comment by fpgamcu** [comment]
It is limiting number of connections in client. I can also implement in bootloader if needed.

**Comment by ralcaide** [comment]
Let me think about that. One question: Is there already a limitation mechanism in the BL, isn't? Or if the BL is not able to connect or download the file stays forever in BL? I think no, after several tries go to the client, isn't? 
Please clarify this, if there is already a limitation mechanism we can avoid a new mechanism

**Comment by fpgamcu** [comment]
Yes, there is limitation mechanism in the BL. It just not synced up with current implemented in FW, but I think it is not needed, because BL is executed only once at device start / restart.

**Comment by ralcaide** [comment]
Great, we will test the current version and if a modification is required in the BL I will let you know through a new issue in BL

**Comment by soporteHW** [comment]
With last FW version V1.46.1 the device doesn't connect to MQTT Broker (GPRS model).

Attached the DEV.TXT file saved in SD memory.
[DEV.TXT](https://github.com/Sinapse-Energia/AP-Sinapse/files/2585220/DEV.TXT)

**Comment by soporteHW** [comment]
With ETH model, MQTT connection is stablished

![image](https://user-images.githubusercontent.com/28917446/48555123-8ebefe00-e8e0-11e8-964c-55351a53e502.png)

**Comment by ralcaide** [comment]
Because the ETH model is still connecting, seems the problem is focused on the new GPRS features.
We are testing with Bootloader 1.30 without deleting the SD. Is maybe related with that? 
@fpgamcu , please take a look to this problem with high priority because at this moment the GPRS version is not working anymore

**Comment by soporteHW** [comment]
@ralcaide The SD is erased. The steps performed were:

1) DUT connected with previous FW version (v1.34 or v1.40.1)
2) Sended 207; command to search for nee FW file .bin
3) Sended 208; to activate BL
4) Sended 314; to erase the NVM and restart the device

**Comment by fpgamcu** [comment]
GPRS version is working ok in my device.
I will check now bootloader and test with it.

**Comment by ralcaide** [comment]
@fpgamcu , in our devices is also working. It was a problem with the binary.

**Comment by soporteHW** [comment]
Tested with the following results:

- Max_connections_hour= 10

![limit_connections_1](https://user-images.githubusercontent.com/28917446/49208752-e23d4b80-f3b8-11e8-91bd-716aa017ab39.png)
![limit_connections_2](https://user-images.githubusercontent.com/28917446/49208759-e6696900-f3b8-11e8-85c5-197888f3c057.png)

Device works as expected. After 10 reboots (sending "398;0;", at <28-11-2018 16:42:39> device doesn't establish MQTT connection till the next hour. At <28-11-2018 17:01:13>, device establishes MQTT connection again.

- Max_connections_day = 50

![limit_50connections_perday](https://user-images.githubusercontent.com/28917446/49208921-6a235580-f3b9-11e8-8558-bf6e059af852.png)

Device works as expected. After 50 reboots (sending "398;0;"), at <28-11-2018 21:24:15> device doesn't establish MQTT connection till the next day.

**Comment by ralcaide** [comment]
Tested OK
